### ISSUE TITLE
[key binding][content assist] Change hotkey for autocomplete selection

### ISSUE THREAD

**Issue by mquan86** [eclipse-platform/eclipse.platform.ui#874]
Copy from: https://bugs.eclipse.org/bugs/show_bug.cgi?id=367009

See http://stackoverflow.com/questions/3663015/change-hotkey-for-autocomplete-selection

In Eclipse, the Enter key is the hotkey that selects an item from the Content Assist list.  Tab might be a better selection hotkey since tab is not likely to be hit mid-line.

Please add the ability to configure which key(s) select an item from Content Assist.

Reproducible: Always

**Comment by mickaelistria** [comment]
Please consider submitting a PR to add such feature.

**Comment by mquan86** [comment]
> 

Thanks for suggestion, I would love to do so but I'm not familiar with Eclipse platform developments so I can't.

**Comment by elsazac** [comment]
Here is a relevant code: 
https://github.com/eclipse-platform/eclipse.platform.ui/blob/369051299e77a841edcabb54df6c9192c4bdb37a/bundles/org.eclipse.jface.text/src/org/eclipse/jface/text/contentassist/CompletionProposalPopup.java#L1392-L1401

However I find couple of challenges while implementing this:
 - The tab already has a behaviour: loop through the controls in the popup. introducing this change will stop that behaviour.
 - we can add a preference to override that behaviour; however the content assist preference belongs to JDT UI
 
 Not sure how to implement this in a clean manner. Any ideas?

**Comment by laeubi** [comment]
> * we can add a preference to override that behaviour; however the content assist preference belongs to JDT UI

I don't think that content assist is a JDT feature...

**Comment by elsazac** [comment]
> > * we can add a preference to override that behaviour; however the content assist preference belongs to JDT UI
> 
> I don't think that content assist is a JDT feature...

I invoked the spy when the content assist is in action and it showed JDT UI

**Comment by elsazac** [comment]
This is what I did or am I missing something?
<img width="813" alt="image" src="https://github.com/eclipse-platform/eclipse.platform.ui/assets/122080218/c9ac1220-efce-4ade-b6f5-a46bb964d98e">

**Comment by laeubi** [comment]
This is the Preference page for the **Java Editor** content assist ... but XML editors, Javascript, even plain text editors have content assist functions.

**Comment by nitind** [comment]
> This is the Preference page for the **Java Editor** content assist ... but XML editors, Javascript, even plain text editors have content assist functions.

Yes, all of which are individually implemented in those editors. As [mentioned](https://github.com/eclipse-platform/eclipse.platform.ui/issues/874#issuecomment-1900356036) by Elsa, Tab is already used as an traversal key for navigating the popup's controls, and a big appeal for Platform is its widespread support for accessibility and internationalization.

**Comment by elsazac** [comment]
Thanks @laeubi for pointers.
@nitind Thanks. so are you proposing that even if tab is not used for the other editors, don't use it for autocomplete hotkey as the tab key has an already standerdised and established meaning?

**Comment by laeubi** [comment]
What I could think of is to user CTRL+ENTER or something like this, in the end the very first thing needed here would be to make the Code not check for a hard coded char but have a (protected) method `isApplyProposal(KeyEvent)` then consumers (including JDT java editor) can extend the class and offer what ever is suitable!
